docs/DeveloperGuide.md

```diff
@@ -1,33 +1,297 @@
+3. **Step 3 - Command Parsing:** In addition the parser will create a `SetActivityGoalCommand` object with the newly 
+   added activity goal attached to it. The command implements the `SetActivityGoalCommand#execute()` operation and is 
+   passed to the AthletiCLI instance.
+4. **Step 4 - Goal Addition:** The AthletiCLI instance executes the `SetActivityGoalCommand` object. The command will 
+   access the data and retrieve the currently stored list of activity goals stored inside it. The new `ActivityGoal` 
+   object is added to the list.
+
+The following sequence diagram shows how the `set-activity-goal` operation works:
+<p  align="center" >
+  <img width="100%" src="images/AddActivityGoal.svg" alt="Sequence Diagram of set-activity-goal"/>
+</p>
+
+Assume that the user has set a goal to run 10km per week and has already tracked two running activities of 5km each.
+The following describes how the goal evaluation works after being invoked by the user, e.g., with a list-activity-goal command:
+
+5. **Step 5 - Goal Evaluation:** The evaluation of the goal is operated by the `ActivityGoal` object. It retrieves the 
+activity list with the two tracked activities from the data and calls the total distance calculation function. It filters the 
+   activity list according to the specified timespan and sports of the goal. The current value obtained by this, 
+   10km in the example, is returned to the `ActivityGoal` object, which then compares it to the target value of the goal. This mechanism is visualized in the following sequence diagram:
+
+<p  align="center" >
+    <img width="100%" src="images/ActivityGoalEvaluation.svg" alt="Sequence Diagram of activity goal evaluation"/>
+</p>
+
+### Sleep Management in AthletiCLI
+
+#### [Implemented] Finding, Adding, Editing, Deleting, Listing Sleep
+
+1. **Input Processing**: The user's input is passed through AthletiCLI to the Parser Class. Examples of user inputs include:
+    - "add-sleep hours/8 datetime/2021-09-01 06:00" for adding sleep.
+    - "edit-sleep 1 hours/8 datetime/2021-09-01 06:00" for editing sleep.
+    - "delete-sleep 1" for deleting sleep.
+    - "list-sleep" for listing all sleep.
+
+2. **Command Identification**: The Parser Class identifies the type of sleep operation and passes the necessary parameters.
+
+3. **Command Creation**: An instance of the corresponding command class is created (e.g., AddSleepCommand, EditSleepCommand, etc.) and returned to AthletiCLI.
+
+4. **Command Execution**: AthletiCLI executes the command, interacting with the data instance of SleepList to perform the required operation.
+
+5. **Result Display**: A message is returned post-execution and passed through AthletiCLI to the UI for display to the user.
+
+The following class diagram shows how sleep and sleep-related classes are constructed in AthletiCLI:
+
+<p  align="center" width="100%">
+  <img width="80%" src="images/SleepAndSleepListClassDiagram.svg" alt="Class Diagram of Sleep and SleepList"/>
```

Comment:
Class diagram is extremely cluttered with methods. Are there any methods which are less important to the overall functionality which can be removed? Doing so might make it easier to read and understand.
![Uploading Screenshot 2023-11-02 at 5.38.42 PM.png…]()